### DISCUSSION TITLE
Is there a way to use an edited "xlgt_04_sm2135" when I compile?

### DISCUSSION THREAD

**Discussion by mallen012** [benzino77/tasmocompiler#426] [Q&A]
Hey everyone, I have about 10 lights that I use Tasmota on, but need to have an edited "xlgt_04_sm2135" file when compiling. Is there an easy way to do this? Using gitpod to accomplish this is very cumbersome, especially when I want to upgrade multiple lights. 

This is a great project, thanks for all of your work on it!

Mike

**Comment by benzino77** [comment]
Hi,

The only possibility you have in TasmoCompoiler is _Custom parameters_ filed where you can put your `#define/#udef` directives.
But...

There is description provided in [ADVANCED.md](https://github.com/benzino77/tasmocompiler/blob/master/ADVANCED.md) how to use custom fork/version of Tasmota. You can prepare your fork/version of Tasmota with modified `xlgt_04_sm2135` file and use TasmoCompiler to compile your, modified version of Tasmota code.

**Reply by mallen012** [reply]
oh, thats great. thanks for the help

**Reply by tve** [reply]
it looks to me like the 'development' branch is only available if one uses a 'dev' version of tasmocompiler, e.g, benzino77/tasmocompiler:development and then one has to check-in any changes before compiling because the compiler forces a checkout.
